### PR TITLE
Lease cleanup v1.x

### PR DESCRIPTION
**Pull request by joshua-kim**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShutdownTask.java [hunk 1]
```diff
@@ -97,47 +109,48 @@ public TaskResult call() {
...
                 }
-            }
 
+                if (!CollectionUtils.isNullOrEmpty(childShards)) {
+                    createLeasesForChildShardsIfNotExist();
+                    updateCurrentLeaseWithChildShards(currentLease);
+                } else {
+                    LOG.warn("Shard " + shardInfo.getShardId()
+                                     + " no longer exists. Shutting down consumer with SHARD_END reason without creating leases for child shards.");
+                }
```
Comment (ashwing) [inline src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShutdownTask.java:133]: Can you incorporate these changes in? https://github.com/ashwing/amazon-kinesis-client/pull/58

File: src/main/java/com/amazonaws/services/kinesis/leases/LeasePendingDeletion.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package com.amazonaws.services.kinesis.leases;
+
```
Comment (ashwing) [inline src/main/java/com/amazonaws/services/kinesis/leases/LeasePendingDeletion.java:2]: License blob

**Review by ashwing — COMMENTED** [review]
Left a comment on incorporating updates from 2.X version. Also we need to rebase the CR once child shards changes are in.

**Comment by joshua-kim** [comment]
Will update PR next week, still having a lot of stuff going on this week with canaries + oncall.

**Comment by joshua-kim** [comment]
Remote branch broke - moving to https://github.com/ashwing/amazon-kinesis-client/pull/73/files

**Pull request closed without merging**